core/modules/user/js/user.js

```diff
@@ -166,9 +174,28 @@ Backdrop.evaluatePasswordStrength = function (password, username) {
   // its length. Again, adapted from zxcvbn.
   strength = (Math.log(cardinality) / Math.log(2)) * password.length + 1;
 
-  // Check if password is the same as the username.
-  if (password !== '' && password.toLowerCase() === username.toLowerCase()) {
-    strength = 5;
+  // Check if password is the same as the username or email.
+  if (password !== '') {
+    if (password === username || password === email) {
+      strength = 5;
+    }
+    // Consider admin password constraint settings if active.
+    if (config.constraints_active === "1") {
+      if (config.no_case === 1) {
+        password = password.toLowerCase();
+        username = username.toLowerCase();
+        email = email.toLowerCase();
+      }
```

Comment:
This should NOT be outside. The previous if clause is for "indicate only".

Reply:
Exactly. The assessment should be the same either way. Regardless of the setting being "indicate" or "enforce", the case (in)sensitivity setting should apply to both. This will be what "grades" the strength of the password. Then we should check whether indicate/enforce, and act accordingly.